### ISSUE TITLE
Very high memory usage with Maestro Studio on Linux

### ISSUE THREAD

**Issue by meatnordrink** [mobile-dev-inc/maestro#1087]
**Describe the bug**
Memory use on Linux increases dramatically when using Maestro Studio - it will slowly and steadily increase from a baseline of about 44% (of 32G) until the computer crashes. Closing Maestro Studio immediately reduces memory usage back to ~44%. (The tests finishing does not affect memory usage.)

It appears that this happens when Maestro Studio is running while a Maestro test on the device it is observing is running, even if Maestro Studio itself is not being interacted with. 

**To Reproduce**

(On Linux)
Launch an emulator with an app built. 
Launch Maestro Studio - `maestro studio`. 
Run a few long Maestro tests. Watch your memory use increase.

**Expected behavior**

Memory use to increase slightly, due to Maestro Studio running, but then remain steady.

**Environment information (please complete the following information):**

- Maestro version: 1.27.0
- Platform: Android
- Framework: React Native
- Device model and OS version: any android (usually Pixel 6 or Nexus S)
- Simulator: Android Studio Emulator
- Host: Linux (Ubuntu)

**Workspace (if applicable)**
I can't share the application code, nor is the issue really about a Maestro test, but if there's specific information you need let me know and I'll do my best.

**Comment by meatnordrink** [comment]
A further note on this - on Linux, killing Maestro Studio in one terminal with `ctrl + c` _also_ kills any simultaneously-running Maestro test. 

I thought this was known, but I just now realized this isn't the case on iOS - and so wondered if it might be connected to the memory leak bug. (Which is still very much happening - Maestro Studio is only usabel for about ~7min on Linux.)

**Comment by synapticvoid** [comment]
I have the same issue with the latest version of Maestro.

**Comment by scriptas** [comment]
I run maestro tests from a raspberry, so this high memory usage is a massive issue. It completely freezes a (32gb sd card) raspberry pi in around 20 hours.

Restarting RPi works, but just for around 24 hours.

After investigating, I found that ./tmp folder is getting jammed up by hundreds of maestro-app*.apk & maestro-server*.apk files. Deleting those will clear up the memory.

As a temporary fix, I will just involve a cleanup of those files as a solution.

**Comment by bartekpacia** [comment]
Hey @scriptas – do you mean that you use Maestro Studio, or just run tests with `maestro test` on your Raspberry Pi? Also, thanks for sharing the findings with us.

**Comment by scriptas** [comment]
@bartekpacia I use raspberry to run the tests. I do not use the maestro studio.

And that fix I've suggested to remove temp files has now been working very well for me for the last 6 months. Raspberry is no longer freezing, and seems like no important data related to the test logs is actually being deleted.

**Comment by meatnordrink** [comment]
Yes, @Fishbowler - just pulled the latest version of the Maestro CLI and ran Maestro Studio against a local simulator. Navigated around a few screens, then left it on while I went to fix tea - when I came back CPU usage was at 100% across all 8 cores, and the computer was nearly frozen. After 5 min of trying was able to close Maestro Studio, and CPU usage returned to ~5%.

**Comment by Fishbowler** [comment]
Thanks! Will investigate!

**Comment by meatnordrink** [comment]
Still happening. Just forgot to kill Studio and it crashed my whole machine.

**Comment by Fishbowler** [comment]
Have given this a go on a fresh Kubuntu 25.04 system. I can see the memory climbing with no interactions. I watched it climb from 410 to 490 in 45 minutes. There's definitely something up in there.
